.gitmodules

```diff
@@ -22,3 +22,12 @@
 [submodule "deps/ringbuf"]
 	path = deps/ringbuf
 	url = https://github.com/agerasev/ringbuf.git
+[submodule "deps/resolv-conf"]
+	path = deps/resolv-conf
+	url = https://github.com/tailhook/resolv-conf.git
+[submodule "deps/quick-error"]
+	path = deps/quick-error
+	url = https://github.com/tailhook/quick-error.git
+[submodule "deps/no-std-net"]
+	path = deps/no-std-net
+	url = https://github.com/dunmatt/no-std-net.git
```

Comment:
Personally, I really found these Git submodules annoying. I would really like to have as few of them as possible. What's more annoying is the patch files for them. Hard to read, hard to maintain.

I am ok with putting `resolv-conf` under the `deps` directory, but not with `quick-error` and `no-std-net` crates. They are only indirect dependencies (Occlum does not depend only them, `resolv-conf` does). But their appearance in `deps` seem to suggest they are direct dependencies. If we keep adding submodules into `deps` this way, the complexity of the directory would quickly get out of control. And the `no-std-net` crate looks especially dubious as its code base is very immature. Thus, depending on it does not seem a good idea to me.

I looked through the patch files. I cannot see why `resolv-conf` cannot be ported with Rust SGX SDK. By using Rust SGX SDK, we can at least get away with `no-std-net` crate (plus the `core-error` crate, which is even worse than `no-std-net` as a dependency).

Reply:
Done